SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/WorkerCoordinator/AbstractWorkerCoordinator.cs

```diff
@@ -51,20 +60,29 @@ protected Process CreateSimulatedPlayerProcess(string fileName, string args)
 
         /// <summary>
         /// Blocks until all active simulated player processes have exited.
+        /// Restarts failed simulated player processes.
         /// Will only wait for processes started through CreateSimulatedPlayerProcess().
         /// </summary>
         protected void WaitForPlayersToExit()
         {
-            while (ActiveProcesses.Count > 0)
+            while (true)
             {
-                try
+                var finishedProcesses = ActiveProcesses.Where(process => process.HasExited).ToList();
+
+                var incorrectlyFinishedProcesses = finishedProcesses.Where(process => process.ExitCode != 0).ToList();
```

Comment:
Very minor: as far as I know we don't have standards for c# but since we are in the 'only auto when shortening very long types' in c++ camp you could argue that we should apply the same logic to var. What I mean is you could consider using the full type here.

Reply:
On second thought, there doesn't seem to be any GDK consensus on this so fine as is.

Reply:
Yeah I know - I'm trying to stick to the code style of other code in the module, and it's very `var`-heavy. Do let me know if you think I should go for full type names as I generally prefer them, too.